environs/simplestreams/simplestreams.go

```diff
@@ -384,12 +384,27 @@ type GetMetadataParams struct {
 	StreamsVersion   string
 	LookupConstraint LookupConstraint
 	ValueParams      ValueParams
-}
-
-// GetMetadata returns metadata records matching the specified constraint,looking in each source for signed metadata.
-// If onlySigned is false and no signed metadata is found in a source, the source is used to look for unsigned metadata.
-// Each source is tried in turn until at least one signed (or unsigned) match is found.
-func GetMetadata(sources []DataSource, params GetMetadataParams) (items []interface{}, resolveInfo *ResolveInfo, err error) {
+	AllResults       bool
+}
+
+type MetadataResult struct {
+	Items       []interface{}
+	ResolveInfo *ResolveInfo
+}
+
+// GetMetadata returns metadata records matching the specified
+// constraint,looking in each source for signed metadata. If
+// onlySigned is false and no signed metadata is found in a source,
+// the source is used to look for unsigned metadata. Each source is
+// tried in turn - if AllResults is false then the results from the
+// first signed (or unsigned) match are returned. If AllResults is
```

Comment:
not so much "match" as the first datasource containing a valid index

Reply:
Thanks, will update.